src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -298,6 +304,17 @@ void drawing_classifier::init_training(
     {"training_iterations", 0}
   });
 
+  auto warm_start = opts.at("warm_start");
+  std::string mlmodel_path;
+  if ( warm_start == true ) {
+    auto mlmodel_path_iter = opts.find("mlmodel_path");
+    if (mlmodel_path_iter == opts.end()) {
+      log_and_throw("Expected option \"mlmodel_path\" not found.");
+    }
+    mlmodel_path = mlmodel_path_iter->second.to<std::string>();
```

Comment:
to me, you don't need to introduce `mlmodel_path` to the outer scope. Putting all relevant logic into the right decision branch is better to group your code.
```
if ( warm_start == true ) {
``` 
Please don't feel strange if I'm too polite today.

Reply:
At this stage in the release, let's avoid changing the order of when initialize `nn_spec_` below, which depends on the value of `mlmodel_path`

Reply:
Honestly, instead of looking at `warm_start`, we could just go whether or not there is a `mlmodel_path` key. Trust the Python side to do the right thing for now, aligning the value of `mlmodel_path` with the value of `warm_start`

Reply:
I'm not sure I follow..

`mlmodel_path` is referenced below so it needs to be defined outside any one scope.

Reply:
done